src/ring_theory/integral_closure.lean

```diff
@@ -123,14 +123,27 @@ theorem is_integral_alg_equiv (f : A ≃ₐ[R] B) {x : A} : is_integral R (f x)
 ⟨λ h, by simpa using is_integral_alg_hom f.symm.to_alg_hom h, is_integral_alg_hom f.to_alg_hom⟩
 
 theorem is_integral_of_is_scalar_tower [algebra A B] [is_scalar_tower R A B]
-  (x : B) (hx : is_integral R x) : is_integral A x :=
+  {x : B} (hx : is_integral R x) : is_integral A x :=
 let ⟨p, hp, hpx⟩ := hx in
 ⟨p.map $ algebra_map R A, hp.map _,
   by rw [← aeval_def, aeval_map_algebra_map, aeval_def, hpx]⟩
 
+lemma is_integral_alg_hom_of_is_scalar_tower [algebra A B] [is_scalar_tower R A B] {b : B}
+  (f : B →ₐ[A] B) (hb : is_integral R b) : is_integral R (f b) :=
```

Comment:
Does this generalize to `f : B →ₐ[A] C`? Is so, could this totally replace `is_integral_alg_hom`?

Reply:
Yes, well spotted! I've generalized `is_integral_alg_hom` and removed `is_integral_alg_hom_of_is_scalar_tower`.